crowbar_framework/lib/crowbar/repository.rb

```diff
@@ -63,94 +97,134 @@ def check_all_repos
         repochecks
       end
 
-      def admin_platform
-        NodeObject.admin_node.target_platform
-      end
+      def provided_and_enabled?(feature, platform = nil)
+        answer = false
 
-      def web_port
-        Proposal.where(barclamp: "provisioner").first.raw_attributes["web_port"]
-      end
+        if platform.nil?
+          all_platforms.each do |p|
+            if provided_and_enabled?(feature, p)
+              answer = true
+              break
+            end
+          end
+        else
+          found = false
+          answer = true
 
-      def repositories(platform)
-        registry[platform]["repos"].keys
+          repositories(platform).each do |repo|
+            provided_features = registry[platform]["repos"][repo]["features"] || []
+
+            next unless provided_features.include? feature
+            found = true
+
+            r = self.new(platform, repo)
+            answer &&= r.active?
+          end
+
+          answer = false unless found
+        end
+
+        answer
       end
 
       def admin_ip
         NodeObject.admin_node.ip
       end
+
+      def web_port
+        Proposal.where(barclamp: "provisioner").first.raw_attributes["web_port"]
+      end
     end
 
     def initialize(platform, repo)
       @platform = platform
       @id = repo
-      @registry = Repository.registry[@platform]["repos"][@id]
-      @url = url
+      @config = Repository.registry[@platform]["repos"][@id]
+    end
+
+    def remote?
+      !@config["url"].blank?
     end
 
     def available?
-      all_repo_dirs.include?(@registry["name"]) && check_repo_tag
+      remote? || (check_directory && check_repo_tag && check_key_file)
     end
 
-    def valid_key_file?
-      if File.exist?(repomd_key_path)
-        md5 = Digest::MD5.hexdigest(File.read(repomd_key_path))
-        repomd_key_md5 == md5
-      else
-        false
-      end
+    def exist?
+      remote? || check_directory
     end
 
-    def repodata_path
-      "/srv/tftpboot/#{@platform}/repos/#{@registry['name']}/repodata"
+    def valid_repo?
+      remote? || check_repo_tag
     end
 
-    def repomd_key_md5
-      @registry["repomd"]["md5"]
+    def valid_key_file?
+      remote? || check_key_file
     end
 
-    def repomd_key_path
-      @registry["repomd"]["key"] || "#{repodata_path}/repomd.xml.key"
+    def name
+      @config["name"] || @id
```

Comment:
related to https://github.com/crowbar/crowbar-core/pull/102#discussion_r42997204:

`@id` is always the key from the yml file which is lower case, `@config["name"]` is the actual uppercase name, so that is not correct here